x-pack/plugins/security_solution/public/common/components/markdown_editor/plugins/insight/index.tsx

```diff
@@ -238,28 +158,48 @@ const InsightComponent = ({
       }),
     });
   }
-  const { data: alertData } = useContext(BasicAlertDataContext);
+  const { data: alertData, timestamp } = useContext(BasicAlertDataContext);
   const { dataProviders, filters } = useInsightDataProviders({
     providers: parsedProviders,
     alertData,
   });
-  const { totalCount, isQueryLoading, oldestTimestamp, hasError } = useInsightQuery({
-    dataProviders,
-    filters,
-  });
-  const timerange: TimeRange = useMemo(() => {
+  const relativeTimerange: TimeRange | null = useMemo(() => {
     if (relativeFrom && relativeTo) {
       const fromStr = relativeFrom;
       const toStr = relativeTo;
-      const from = parseDateWithDefault(fromStr, DEFAULT_FROM_MOMENT).toISOString();
-      const to = parseDateWithDefault(toStr, DEFAULT_TO_MOMENT, true).toISOString();
+      const alertRelativeDate = timestamp ? moment(timestamp) : moment();
+      const from = parseDateWithDefault(
+        fromStr,
+        DEFAULT_FROM_MOMENT,
+        false,
+        moment,
+        alertRelativeDate.toDate()
+      ).toISOString();
+      const to = parseDateWithDefault(
+        toStr,
+        DEFAULT_TO_MOMENT,
+        true,
+        moment,
+        alertRelativeDate.toDate()
+      ).toISOString();
       return {
-        kind: 'relative',
+        kind: 'absolute',
         from,
         to,
-        fromStr,
-        toStr,
       };
+    } else {
+      return null;
+    }
+  }, [relativeFrom, relativeTo, timestamp]);
+
+  const { totalCount, isQueryLoading, oldestTimestamp, hasError } = useInsightQuery({
+    dataProviders,
+    filters,
+    relativeTimerange,
+  });
+  const timerange: TimeRange = useMemo(() => {
+    if (relativeTimerange) {
```

Comment:
similar thing here (feel free to ignore also), I feel like the if-else if-else is unnecessary and makes it a bit harder to read. Just have 2 if conditions and return every time

Reply:
well this field is optional, and also not guaranteed to be a valid date, as users can set it in advanced settings, and typos would cause invalid dates. Plus, if the user supplies the override, we want to use that everytime, but if not, we  determine the oldestTimestamp and use that to make investigate in timeline time bounded relative to the alert, to help with performance, but not using the possibly irrelevant default of now - 30 days.

Reply:
I just meant that it could be written like this (which seems easier on the eye to me)
```
if (relativeTimerange) {
      return relativeTimerange;
}

if (oldestTimestamp != null) {
      return {
        kind: 'absolute',
        from: oldestTimestamp,
        to: new Date().toISOString(),
      };
}

const { to, from, fromStr, toStr } = getTimeRangeSettings();
return {
      kind: 'relative',
      to,
      from,
      fromStr,
      toStr,
};
```